### PR TITLE
Allow kwargs in network creation to override Class-level variables for Roadway and Transit

### PR DESCRIPTION
**Pull request by e-lo**
This PR adds steps to the `read()` function for both RoadwayNetwork and TransitNetwork which will read in kwargs from the `read()` variables and will update any matching (upper-case) class-level variables to the specified values.

This is useful for when an agency might need to use a different unique ID or variable set from the defaults. 

Also:
- added util method: `class_vars()`
- test network links which use different Key than `model_link_id`

Tested via: `test_kwarg_instantiation()`

Closes Issue #290

### CODE REVIEW

File: tests/test_roadway.py [hunk 1]
```diff
@@ -1244,3 +1243,24 @@ def test_managed_lane_restricted_access_egress(request):
...
+@pytest.mark.menow
+def test_kwarg_instantiation(request):
+    WranglerLogger.info(f"--Starting: {request.node.name}")
+
+    update_values = {
+        'unique_link_key':'different_link_id'
+    }
+
+    net = RoadwayNetwork.read(
+        link_file=SMALL_LINK_FILE,
```
Comment (i-am-sijia) [inline tests/test_roadway.py:1256]: Shouldn't we be testing with the link_diffIDs.json?

File: network_wrangler/roadwaynetwork.py [hunk 2]
```diff
@@ -163,17 +163,17 @@ class RoadwayNetwork(object):
 
     UNIQUE_LINK_KEY = "model_link_id"
     UNIQUE_NODE_KEY = "model_node_id"
-    UNIQUE_MODEL_LINK_IDENTIFIERS = ["model_link_id"]
+    UNIQUE_MODEL_LINK_IDENTIFIERS = [UNIQUE_LINK_KEY]
```
Comment (i-am-sijia) [inline network_wrangler/roadwaynetwork.py:166]: In the example test [`test_kwarg_instantiation()`](https://github.com/wsp-sag/network_wrangler/blob/06740ad41b26418c7ef1276260fc9e59c9b0df0f/tests/test_roadway.py#L1252C8-L1252C8) in test_roadway.py, we update the `UNIQUE_LINK_KEY` to `"different_link_id"`, which works, but it does not automatically update the reference of `UNIQUE_LINK_KEY` in the class, i.e., the `UNIQUE_MODEL_LINK_IDENTIFIEDS` here would still be `["model_link_id"]`, which is causing the test using "links_DiffIDs.json" to fail.

We need to refresh all class attrs whenever setattr() is called.

File: network_wrangler/transitnetwork.py [hunk 3]
```diff
@@ -125,6 +128,19 @@ def read(feed_path: str) -> TransitNetwork:
...
+        """
+        Update Class Variables
+        """
+        _class_vars = class_vars(TransitNetwork)
+        for k,v in kwargs.items():
+            if k.upper() not in _class_vars:
+                WranglerLogger.warning(f"{k} specified in TransitNetwork kwargs but does not match a Class-level variable.")
+                continue
+            WranglerLogger.debug(f"Updating TransityNetwork.{k.upper()} to: {v}")
+            setattr(RoadwayNetwork,k.upper(),v)
```
Comment (i-am-sijia) [inline network_wrangler/transitnetwork.py:140]: Here for the same issue in roadwaynetwork.py

File: network_wrangler/roadwaynetwork.py [hunk 4]
```diff
@@ -258,7 +260,18 @@ def read(
...
+        """
+        Update Class Variables
+        """
+        _class_vars = class_vars(RoadwayNetwork)
+        for k,v in kwargs.items():
+            if k.upper() not in _class_vars:
+                WranglerLogger.warning(f"{k} specified in RoadwayNetwork kwargs but does not match a Class-level variable.")
+                continue
+            WranglerLogger.debug(f"Updating RoadwayNetwork.{k.upper()} to: {v}")
+            setattr(RoadwayNetwork,k.upper(),v)
```
Comment (i-am-sijia) [inline network_wrangler/roadwaynetwork.py:272]: after this setattr(), can we refresh all the attrs?

File: tests/test_roadway.py (hunk 1 above)
Comment (i-am-sijia) [reply tests/test_roadway.py:1256]: Test fails when using the link_diffIDs.json. Error:

```python
2023-07-06 15:20:19, ERROR: Network doesn't contain unique link identifier: model_link_id
```

**Review by i-am-sijia — CHANGES_REQUESTED** [review]

**Comment by e-lo** [comment]
Closing this as this approach has been superceeded by aligning on a string model for IDs across all tables in the `selection-refactor` branch.

**Pull request closed without merging**